nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java

```diff
@@ -567,40 +565,29 @@ public void testGCEnabled() {
   }
 
   private String getTableBasePath(String tableName) {
-    String databasePath = temp.toString() + "/" + DB_NAME;
-    return Paths.get(databasePath, tableName).toAbsolutePath().toString();
-  }
-
-  protected Path getTableLocationPath(String tableName) {
-    return new Path("file", null, Paths.get(getTableBasePath(tableName)).toString());
-  }
-
-  protected String getTableLocation(String tableName) {
-    return getTableLocationPath(tableName).toString();
-  }
-
-  private String metadataLocation(String tableName) {
-    return Paths.get(getTableBasePath(tableName), "metadata").toString();
+    return temp.toURI() + DB_NAME + "/" + tableName;
   }
 
   @SuppressWarnings(
       "RegexpSinglelineJava") // respecting this rule requires a lot more lines of code
-  private List<String> metadataFiles(String tableName) {
-    return Arrays.stream(Objects.requireNonNull(new File(metadataLocation(tableName)).listFiles()))
+  private List<String> metadataFiles(File tablePath) {
+    return Arrays.stream(
+            Objects.requireNonNull(
+                new File((tablePath + "/" + "metadata").replace("file:", "")).listFiles()))
```

Comment:
This looks like a safe way to work with strings, but not with files. It looks like you're still fighting the representation used everywhere. Here, you convert file to string, do custom manipulation, and go back to file. That seems needlessly complicated and has implications for readability and correctness (what does `new File(...)` return that requires `"file:"` to need removed??).

Reply:
When `Table.location()` is called, it returns a string with `file:` scheme. So, while creating a `new File()` for a `ListFiles()`, if the `pathname` has the file scheme, `ListFiles` return null. Hence, removing the file scheme. Similar to how it is in many existing test cases. 

But instead of doing for every time, now I moved this logic to the `tablePath` initialization